### PR TITLE
Added listener to restart server when it is actually stopped #86

### LINKED ISSUE #86: Restart server is not working
I faced this problem while testing the activity bar. The server stops but it doesn't restart. The stop command is too slow we need to add a listener to check the server state or something like that..

STEPS:
ASSERT: have wildfly 17.0.1 server running in run mode
EXEC: select "Restart in debug mode"

Result:
Server stops and it doesn't restart.

Same thing if i select "Restart in run mode".

### PR DESCRIPTION
**Pull request by lstocchi**
it fixes #86

### CODE REVIEW

File: src/extensionApi.ts [hunk 1]
```diff
@@ -262,16 +262,28 @@ export class CommandHandler {
...
+            return Promise.reject('Failed to contact the RSP server.');
+        }
+
+        const listener = (state: Protocol.ServerState) => {
+            if (state
+                && state.server
+                && state.server.id === context.server.id
+                && state.state === ServerState.STOPPED) {
+                client.getIncomingHandler().removeOnServerStateChanged(listener);
                 if (mode === 'debug') {
```
Comment (adietish) [inline src/extensionApi.ts]: maybe a switch would be more expressive than the sequence of if/else if/else

Comment (adietish) [reply src/extensionApi.ts]: furthermore I'd also highly encourage defining the mode as a constant. I see 'debug' and 'run' being used and repeated in several place. Once we'd change a mode, we'd have correct all these occurrences

File: test/commands.test.ts [hunk 2]
```diff
@@ -742,41 +740,38 @@ suite('Command Handler', () => {
...
 
-        test('error if mode doesn\'t contains a valid value', async () => {
-            try {
-                await handler.restartServer('fakeMode', ProtocolStubs.startedServerState);
-                expect.fail();
-            } catch (err) {
-                expect(err).equals('Could not restart server: unknown mode fakeMode');
-            }
-        });
+        // test('error if mode doesn\'t contains a valid value', async () => {
```
Comment (adietish) [inline test/commands.test.ts]: this test seem useful, it asserts what happens if the mode is something that differs from 'debug' or 'run'. Why commenting it?

**Comment by odockal** [comment]
@lstocchi Cool, works for me as well, wf 17 + jboss eap 7.1.

**Comment by lstocchi** [comment]
@adietish added switch + fixed tests. Give it a look whenever you can.

I also found out there was a problem with the error messages. If debugServer or startServer failed for some reason, error messages didn't get displayed. Fixed it.

**Pull request merged**